### PR TITLE
Add a joint score test function

### PR DESCRIPTION
**Pull request by amstilp**
Add a function to perform a joint score test using a null model and a matrix of genotypes.

### CODE REVIEW

**Comment by amstilp** [comment]
Does the version number need to be incremented in this branch, or elsewhere?

File: R/jointScoreTest.R [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  betaCov <- solve(GG)
+  rownames(betaCov) <- colnames(betaCov) <- colnames(G)
+
+  # Fixed effect estimates for the variants.
+  beta <- crossprod(betaCov, GY)
+  se <- sqrt(diag(betaCov))
+
+  # Joint test statistic. Note this is the equivalent of the stat being squared,
+  # by convention.
+  Stat.joint <- as.vector(crossprod(GY, beta))
```
Comment (mconomos) [inline R/jointScoreTest.R]: I don't think this is a problem, but Stat.joint is just a scalar. I think could be as.numeric() instead

File: R/jointScoreTest.R [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  pval.joint <- pchisq(Stat.joint, df = ncol(G))
```
Comment (mconomos) [inline R/jointScoreTest.R]: I think this needs to have lower.tail = FALSE

File: R/jointScoreTest.R [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  # Percentage of variance explained jointly by these variants.
+  pve <- as.vector(Stat.joint / nullmod$RSS0)
```
Comment (mconomos) [inline R/jointScoreTest.R]: again a scalar

File: man/jointScoreTest.Rd [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  Colnames for \code{G} are not required.
+  If present, the column names of \code{G} will be used as the rownames of the \code{fixef} element and the column and rownames of the \code{betaCov} element of the output.
+  \code{fixef} and \code{betaCov} will be in the same order as the columns in \code{G}.
+
+  Missing data in \code{G} are not allowed.
+}
+\value{\code{jointScoreTest} returns a list with the following elements:
+  \item{Stat.joint}{Squared joint score test statistic for all variants in \code{G}.}
+  \item{pval.joint}{p-value associated with the joint score test statistic drawn from a \deqn{\chi^2} distribution with \eqn{n_\mathrm{variants}} degrees of freedom.}
+  \item{pve}{Estimate of the proportion of variance explained by the variants in \code{G}.}
```
Comment (mconomos) [inline man/jointScoreTest.Rd]: small tweak to language:

Estimate of the proportion of variance explained jointly by the variants in \code{G}. 

Also, should we call this `pve.joint` to match the Stat and pval?

**Review by mconomos — CHANGES_REQUESTED** [review]
Noticed a couple of minor things to address; left in comments. Sorry I missed them before.

File: man/jointScoreTest.Rd (hunk 4 above)
Comment (amstilp) [reply man/jointScoreTest.Rd]: Yep, I think that makes sense -- changed to `pve.joint`.

**Comment by amstilp** [comment]
Thanks @mconomos! I made all these changes.

**Review by smgogarten — CHANGES_REQUESTED** [review]

File: man/jointScoreTest.Rd [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+rownames(phen) <- rownames(geno)
+
+# load saved matrix of KING-robust estimates
+data("HapMap_ASW_MXL_KINGmat")
+cov.mat = HapMap_ASW_MXL_KINGmat[rownames(phen), rownames(phen)]
+
+# Fit a null model.
+nullmod <- fitNullModel(phen,
+                        outcome = "outcome",
+                        "covars" = "covar",
```
[the remark this answers is not in the document]
Comment (amstilp) [reply man/jointScoreTest.Rd]: Yes! Fixed.
